Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1318 @@
+
+            _interfaceAddresses = new Collection<IPObject>();
+            _macAddresses = new List<PhysicalAddress>();
+            _interfaceNames = new Dictionary<string, int>();
+            _publishedServerUrls = new Dictionary<IPNetAddress, string>();
+            _eventFireLock = new object();
+
+            UpdateSettings(_configurationManager.GetNetworkConfiguration());
+
+            NetworkChange.NetworkAddressChanged += OnNetworkAddressChanged;
+            NetworkChange.NetworkAvailabilityChanged += OnNetworkAvailabilityChanged;
+
+            _configurationManager.NamedConfigurationUpdated += ConfigurationUpdated;
+        }
+#pragma warning restore CS8618 // Non-nullable field is uninitialized.
+
+        /// <summary>
+        /// Event triggered on network changes.
+        /// </summary>
+        public event EventHandler? NetworkChanged;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether testing is taking place.
+        /// </summary>
+        public static string MockNetworkSettings { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether IP6 is enabled.
+        /// </summary>
+        public bool IsIP6Enabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether IP4 is enabled.
+        /// </summary>
+        public bool IsIP4Enabled { get; set; }
+
+        /// <inheritdoc/>
+        public Collection<IPObject> RemoteAddressFilter { get; private set; }
+
+        /// <summary>
+        /// Gets a value indicating whether is all IPv6 interfaces are trusted as internal.
+        /// </summary>
+        public bool TrustAllIP6Interfaces { get; internal set; }
+
+        /// <summary>
+        /// Gets the Published server override list.
+        /// </summary>
+        public Dictionary<IPNetAddress, string> PublishedServerUrls => _publishedServerUrls;
+
+        /// <summary>
+        /// Creates a new network collection.
+        /// </summary>
+        /// <param name="source">Items to assign the collection, or null.</param>
+        /// <returns>The collection created.</returns>
+        public static Collection<IPObject> CreateCollection(IEnumerable<IPObject>? source = null)
+        {
+            var result = new Collection<IPObject>();
+            if (source != null)
+            {
+                foreach (var item in source)
+                {
+                    result.AddItem(item);
+                }
+            }
+
+            return result;
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <inheritdoc/>
+        public IReadOnlyCollection<PhysicalAddress> GetMacAddresses()
+        {
+            // Populated in construction - so always has values.
+            return _macAddresses.AsReadOnly();
```

Comment:
We never use `AsReadOnly` as we trust the callers

Reply:
Removed it.